astropy/time/core.py

```diff
@@ -1262,6 +1262,80 @@ def isclose(self, other, atol=None):
 
         return out
 
+    def __replace__(self, /, **changes):
+        for key, val in {
+            "jd1": self._time.jd1,
+            "jd2": self._time.jd2,
+            "scale": self.scale,
+            "format": self.format,
+            "precision": self.precision,
+            "in_subfmt": self.in_subfmt,
+            "out_subfmt": self.out_subfmt,
+            "delta_ut1_utc": getattr(self, "_delta_ut1_utc", None),
+            "delta_tdb_tt": getattr(self, "_delta_tdb_tt", None),
+        }.items():
+            if key not in changes:
+                changes[key] = copy.copy(val)
```

Comment:
So, the main problem is here: `copy.copy(self._time.jd1)` is not the same as `self._time.jd1.copy()` because if calls `.__copy__()` instead of `.copy()`. So, in the previous incarnation, one could not so easily get a masked array with no base.

More generally, this new `__replace__` duplicates a *lot* of `_apply`...

Need to think a bit more on how to solve this all, but now need to run...